[WIP] [DOCS] Add Nebari installation guides and getting-started as first directions (#45)
nebari-dev/nebari: docs/docs/started/deploy.md

---
id: deploy
---

> # Deploying options

Reviewer: What are we trying to refer to here? If not cloud platforms, then what?

> # Deploying options

Reviewer: @kcpevey What are we trying to refer to here? If not cloud platforms, then what?

> # Deploying options

Reviewer: Looks like maybe: cloud vs HPC vs local?

> # Deploying options

Reviewer 2: I think it refers to "which cloud platform" AND? "hpc vs local". I'd say leaving it as `Choosing a deployment platform` will be the most appropriate.

A guide to help deploying Nebari with best suited option.

Nebari can be deployed on a bare-metal server using HPC or on a Cloud provider. Review the options below to discover which option best suits your needs.

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs>
  <TabItem value="cloud" label="Cloud" default>

The Cloud deployment of Nebari is considered to be the default option. It enables teams to build and maintain a cost effective and scalable compute/data science platform in the Cloud, providing an [Infrastructure as Code](https://en.wikipedia.org/wiki/Infrastructure_as_code) platform that streamlines the deployment of data science infrastructure.

If you are not sure which option to choose, try this one. It is suitable for most use cases, especially if:

- You require scalable infrastructure

> - You aim to have a production environment with GitOps enabled by default

Author: https://docs.qhub.dev/en/stable/source/installation/setup.html?#ci-cd-pipeline

> The cloud installation is based on Kubernetes, but knowledge of Kubernetes is NOT required nor is in depth knowledge about the specific provider required either. QHub supports [Amazon AWS](https://docs.qhub.dev/en/latest/source/installation/setup.html#amazon-web-services-aws), [DigitalOcean](https://docs.qhub.dev/en/latest/source/installation/setup.html#digital-ocean), [Google GCP](https://docs.qhub.dev/en/latest/source/installation/setup.html#google-cloud-platform), and [Azure](https://docs.qhub.dev/en/latest/source/installation/setup.html#microsoft-azure).

Reviewer 2: update links to the relative pages - can just add a TODO for now.

</TabItem>
<TabItem value="hpc" label="HPC">

> Nebari HPC is an opinionated open source deployment of jupyterhub based on an HPC jobscheduler. Nebari HPC is a "distribution" of these packages much like Debian and Ubuntu are distributions of Linux. The high level goal of this distribution is to form a cohesive set of tools that enable:

Reviewer 2: Its probably worth adding a note to say that `HPC` is actually more generic. I dont have the vocabulary to know the right words on how to explain, but its not _just_ for HPCs. That should be explained.

> Nebari HPC is an opinionated open source deployment of jupyterhub based on an HPC jobscheduler. Nebari HPC is a "distribution" of these packages much like Debian and Ubuntu are distributions of Linux. The high level goal of this distribution is to form a cohesive set of tools that enable:

Reviewer: Maybe something like:

> To note, Nebari HPC can be used on other distributed compute hardware, not just HPC hardware specifically. We anticipate that Nebari HPC will be used most often on HPC hardware, however.

---
id: deploy
---

# Deploying options

A guide to help deploying Nebari with best suited option.

> Nebari can be deployed on a bare-metal server using HPC, on a Cloud provider or on a pre-existing kubernetes cluster. Review the options below to discover which option best suits your needs.

Reviewer: ```suggestion
Nebari can be deployed on a bare-metal server using HPC, on one of multiple major cloud providers, or on a pre-existing Kubernetes cluster. Review the options below to determine which option best suits your needs.
```

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs>
  <TabItem value="cloud" label="Cloud" default>

> The Cloud deployment of Nebari is considered to be the default option. It enables teams to build and maintain a cost effective and scalable compute/data science platform in the Cloud, providing an [Infrastructure as Code](https://en.wikipedia.org/wiki/Infrastructure_as_code) platform that streamlines the deployment of data science infrastructure.

Reviewer: ```suggestion
Cloud deployment of Nebari is considered to be the default option. It enables teams to build and maintain a cost effective and scalable compute/data science platform in the cloud, providing an [Infrastructure as Code](https://en.wikipedia.org/wiki/Infrastructure_as_code) platform that streamlines the deployment and management of data science infrastructure.
```

> If you are not sure which option to choose, try this one. It is suitable for most use cases, especially if:

Reviewer: ```suggestion
If you are not sure which option to choose, a cloud installation is likely your best option. It is suitable for most use cases, especially if:
```

- You require scalable infrastructure

- You aim to have a production environment with GitOps enabled by default

The cloud installation is based on Kubernetes, but knowledge of Kubernetes is NOT required nor is in depth knowledge about the specific provider required either. QHub supports [Amazon AWS](https://docs.qhub.dev/en/latest/source/installation/setup.html#amazon-web-services-aws), [DigitalOcean](https://docs.qhub.dev/en/latest/source/installation/setup.html#digital-ocean), [Google GCP](https://docs.qhub.dev/en/latest/source/installation/setup.html#google-cloud-platform), and [Azure](https://docs.qhub.dev/en/latest/source/installation/setup.html#microsoft-azure).

> For instructions on installing and deploying Nebari on a Clod provider, please visit ou [Cloud providers](/started/cloud-providers) for a list of the supported cloud providers and their respectful installation guides.

Reviewer: ```suggestion
For instructions on installing and deploying Nebari on a particular cloud provider, please visit our [cloud providers page](/started/cloud-providers) for a list of the supported cloud providers and their respective installation how-to guides.
```

</TabItem>
<TabItem value="hpc" label="HPC">

Nebari HPC is an opinionated open source deployment of jupyterhub based on an HPC jobscheduler. Nebari HPC is a "distribution" of these packages much like Debian and Ubuntu are distributions of Linux. The high level goal of this distribution is to form a cohesive set of tools that enable:

> - environment management via conda and conda-store

Reviewer: ```suggestion
- Environment management via conda and conda-store
```

- monitoring of compute infrastructure and services

> - scalable and efficient compute via jupyterlab and dask

Reviewer: ```suggestion
- Scalable and efficient compute via JupyterLab and Dask
```

> - deployment of jupyterhub on prem without requiring deep devops knowledge of the Slurm/HPC and jupyter ecosystem

Reviewer: ```suggestion
- On-prem deployment of JupyterHub without requiring deep DevOps knowledge of the Slurm/HPC and Jupyter ecosystems
```

The Nebari HPC should be your choice if:

> - You have highly optimized code that require highly performant infrastructure

Reviewer: ```suggestion
- You have highly optimized code that requires highly performant infrastructure
```

- You have existing infrastructure already available

> - You expect that your infrastructure will not exceed the existing resources capabilities

Reviewer: ```suggestion
- You expect that your projects will not exceed the resources/capabilities of your current infrastructure
```

For instructions on installing and deploying Nebari HPC, please visit [How to install and setup Nebari HPC on bare metal machines](/how-tos/nebari-hpc).

:::note

> Although it is possible to deploy Nebari HPC on the Cloud, it is not generally recommended due to possible high costs. For more information, check out the base cost section of the docs.

Reviewer: Need to add link here to the 'base costs' section of the docs.

Reviewer 2: we can open an issue to come back to this when those docs exist.

:::

</TabItem>
<TabItem value="local" label="Pre-existing">

> The pre-exisitng (or local) version is recommended for **testing and development** of Nebari’s components due to its simplicity. Choose the local mode if:

Author: The local installation is a `pre-existing` one as well because you need to create the Kubernetes ecosystem using Mikikube and then deploy Nebari inside the virtual `node`. But I agree that installing on a pre-existing Kubernetes cluster should go to a different `how-to` as well because it's restricted to what cloud provider or where the cluster is currently hosted/built-in, as well as the security policies already present in the target infrastructure.

> The pre-exisitng (or local) version is recommended for **testing and development** of Nebari’s components due to its simplicity. Choose the local mode if:

Reviewer 3: @kcpevey  did you open an issue?
For now I made a suggestion to focus on "Existing k8s clusters"

- You already have Kubernetes clusters
- You want to test these Kubernetes clusters
- You have available local compute setup
- You want to try out Nebari with a quick-install to see how it works, without setting up environment variables

> You should choose another installation option if you are starting from scratch (i.e. no clusters yet) and aim to have a production environment.

Reviewer: ```suggestion
You should choose another installation option, likely a [cloud install](/started/cloud-providers.md), if you are starting from scratch (you have no compute clusters already established) and you desire to stand up a production instance of Nebari.
```
